### DISCUSSION TITLE
Inconsistent image generated from original pdf

### DISCUSSION THREAD

**Discussion by Yichen-fqyd** [pymupdf/PyMuPDF#1105] [Q&A]
Hi I am using the following function to convert each pdf page to image

```
def page_to_img(page, pdf_zoom = 1):

    pix = page.getPixmap(matrix = fitz.Identity * pdf_zoom)

    mode = "RGBA" if pix.alpha else "RGB"

    img = Image.frombytes(mode, [pix.width, pix.height], pix.samples)

    return img 
``` 

However, there are two pdfs that the converted image is not same with original pdf
For example, the original pdf showed `000000`, while the converted image showed `222222`

I am wondering the reason for this and how to solve this problem?

Thanks in advance

**Comment by JorjMcKie** [comment]
No idea what may be happening there. Weird. I am afraid you would need to send me the PDF page in question.
BTW your code always will produce RGB images - it is the `get_pixmap()` which controls the inclusion of an alpha channel.

How do you save the PIL image once you have created it with the function?

**Reply by Yichen-fqyd** [reply]
I simplely do img_PIL.save("pdf.png") just to check the converted image. Since I was trying to crop the corresponding bbox for ocr for some unrecognized chars. However, I found the inconsistent issue.

(For privacy issue, I send the documents via email, can you help to take a look?

**Reply by JorjMcKie** [reply]
> (For privacy issue, I send the documents via email, can you help to take a look?

Sure, go ahead. You have my e-mail.

**Comment by JorjMcKie** [comment]
Found your PDFs in my spam folder ...
These are weird examples! I can confirm your observations, The error occurs outside PyMuPDF and can be reproduced using the MuPDF CLI tool `mutool` via command ``mutool convert -o page%d.png CourtesyTradeNotification3.pdf``.
You should submit an issue to MuPDF's issue system https://bugs.ghostscript.com/enter_bug.cgi together with a sample PDF and the above command.
Both PDFs definitely are irregular in that they use non-embedded fonts (Arial,bold) with "identity" encoding. This happens exactly in the places where you observed "arbitrary" character replacements ...
Other tools, like Adobe, or PDF-XChange produce "correct" images when using their "snapshot" subtools.
MuPDF is consequent in so far as its own PDF reader does display the PDFs exactly the way as the pixmaps are produced.
So the PDFs' making (as described) do leave room for misinterpretation - which is always bad.
